src/high-level/src/KinDynComputations.cpp

```diff
@@ -844,8 +905,54 @@ bool KinDynComputations::setRobotState(const Transform& world_T_base,
 
     this->invalidateCache();
 
-    // Save pos
-    this->pimpl->m_pos.worldBasePos() = world_T_base;
+    // Handle transformation when floating base is a frame vs a link
+    Transform world_T_baseLink;
+    Twist baseLinkVelocity;
```

Comment:
Can you explain what this variable contains in a comment before the variable is introduced or in the variable name? If the base is a link frame and not an additional frame, this is exactly equal to input velocity (that is expressed in the selected velocity representation), while if the base is an additional frame, this contains the velocity of the link to which the frame is attached, expressed with the velocity representation specified in `m_frameVelRepr` . Can you explain in a comment this, and perhaps name the variable `baseLinkVelocityInSelectedVelRepr` ?